app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -834,10 +950,20 @@ public Cursor call() throws Exception {
         }
     };
 
+    Callable<Cursor> streamsGeneratorWithFilter = new Callable<Cursor>() {
```

Comment:
We could achieve the results using one Callable<Cursor> as well, 
If filter editText is not empty then append the WHERE clauses in the SQL query and return the filtered result else no filtering!